nf-core-mtbseq/modules/local/mtbseq/tbamend.nf

```diff
@@ -0,0 +1,53 @@
+process MTBSEQ_TBAMEND {
+    tag "$meta.id"
+    label 'process_high'
+    conda "bioconda::mtbseq=1.0.4"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/YOUR-TOOL-HERE':
+        'quay.io/biocontainers/YOUR-TOOL-HERE' }"
+
+    input:
+
+    path("Joint/*")
+    path(samplesheet_tsv)
+    env(USER)
+    tuple path(ref_resistance_list), path(ref_interesting_regions), path(ref_gene_categories), path(ref_base_quality_recalibration)
+    output:
+    path("Amend/*"), emit: samples_amended
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    """
+    mkdir Amend
+
+    MTBseq --step TBamend \\
+            --threads ${task.cpus} \\
+            --samples ${samplesheet_tsv} \\
+            --project ${params.project} \\
+            --mincovf ${params.mincovf} \\
+            --mincovr ${params.mincovr} \\
+            --minphred ${params.minphred} \\
+            --minfreq ${params.minfreq} \\
+            --unambig ${params.unambig} \\
+            --window ${params.window} \\
+            --distance ${params.distance} \\
+            --resilist ${ref_resistance_list} \\
+            --intregions ${ref_interesting_regions} \\
+            --categories ${ref_gene_categories} \\
+            --basecalib ${ref_base_quality_recalibration} \\
```

Comment:
Instead of this, I'm thinking that we should rely upon the `task.ext.args` pattern to provide non-file options.

For example 

```nextflow
def args = task.ext.args ?: "--mincovf 4 --mincovr 4 --minphred 4 --minfreq 75 --unambig 95 --window 12 --distance 12 "

MTBseq --step TBamend \\
            --samples ${samplesheet_tsv} \\
            --resilist ${ref_resistance_list} \\
            --intregions ${ref_interesting_regions} \\
            --categories ${ref_gene_categories} \\
            --basecalib ${ref_base_quality_recalibration} 
            ${args} \\
            --threads ${task.cpus} \\
```

Reply:
@abhi18av, that's make complete sense. I'll implement that and return.

Reply:
@Mxrcon , any progress on this one?

Reply:
@abhi18av yes, I've just added theoption args on the modules that could access optional args like `minf`